### PR TITLE
Improve the doc of bitcoin staking script

### PR DESCRIPTION
**Pull request by aaronisme**
Hello Babylon team,

I am really impressed by the work you are doing. Currently, I am investigating how Bitcoin staking transactions work and I found this document. I believe there are some improvements that can be made. Here are two points I've added:
1. A high-level introduction about who will be involved in the Bitcoin staking transaction.
2. I noticed the terms `btc holder` and `btc staker` are used interchangeably. I have standardized the terminology to `btc staker` for consistency.

Please consider these revisions.

### CODE REVIEW

File: docs/staking-script.md [hunk 1]
```diff
@@ -30,6 +30,18 @@ The protocol has the following important properties:
...
    multiple PoS chains and earn multiple PoS yields
 
+
+In the entire Bitcoin staking process, two parties are involved: one is called 
+the Bitcoin Staker, and the other is called the Finality Provider.
+
+- **Bitcoin Staker**: A Bitcoin Staker is a person who owns Bitcoin and chooses 
+to lock up their Bitcoin for a set period of time to earn yields.
+- **Finality Provider**: A Finality Provider is the service that provides security 
+assurance to the proof-of-stake chain.
```
Comment (vitsalis) [inline docs/staking-script.md:40]:
```suggestion
- **Bitcoin Staker**: A Bitcoin Staker is a person who owns Bitcoin and chooses 
to lock up their Bitcoin for a set period of time to secure a PoS chain.
- **Finality Provider**: A Finality Provider is an entity that votes in the finality round
to provide security assurance to the PoS chain.
```

File: docs/staking-script.md [hunk 2]
```diff
@@ -30,6 +30,18 @@ The protocol has the following important properties:
...
+
+The Bitcoin Staker can choose a specific Finality Provider with 
+which to lock (stake) their Bitcoins."
```
Comment (vitsalis) [inline docs/staking-script.md:43]:
```suggestion
The Bitcoin Staker can choose a specific Finality Provider to which 
to they can delegate (stake) their locked Bitcoin."
```

**Review by vitsalis — COMMENTED** [review]
Thanks for this! Only a couple wording suggestions.

**Comment by aaronisme** [comment]
@vitsalis Thanks for the feedback, I have made the change.

File: docs/staking-script.md [hunk 3]
```diff
@@ -30,6 +30,18 @@ The protocol has the following important properties:
...
+In the entire Bitcoin staking process, two parties are involved: one is called 
+the Bitcoin Staker, and the other is called the Finality Provider.
+
+- **Bitcoin Staker**: A Bitcoin Staker is a person who owns Bitcoin and chooses 
+to lock up their Bitcoin for a set period of time to secure a PoS chain.
+- **Finality Provider**: A Finality Provider is the an entity that votes 
+in the finality round to provide security assurance to the PoS chain.
+
+The Bitcoin Staker can choose a specific Finality Provider to which 
+to they can delegate (stake) their locked Bitcoin.
```
Comment (vitsalis) [inline docs/staking-script.md:43]:
```suggestion
they can delegate the voting power stemming from their locked Bitcoin.
```

Comment (vitsalis) [reply docs/staking-script.md:43]: Don't want this to be confused to delegating custody of the locked BTC.

**Review by vitsalis — COMMENTED** [review]
An additional small comment

File: docs/staking-script.md (hunk 3 above)
Comment (aaronisme) [reply docs/staking-script.md]: Rewrite the sentence on the last commit

**Comment by aaronisme** [comment]
> An additional small comment

I have made the change, Thanks for the comment

**Review by vitsalis — APPROVED** [review]
Thank you!

**Pull request merged**